src/NServiceBus.Core/Pipeline/ConnectorContextExtensions.cs

```diff
@@ -163,6 +205,8 @@ public static IOutgoingPhysicalMessageContext CreateOutgoingPhysicalMessageConte
         /// </summary>
         public static IAuditContext CreateAuditContext(this ForkConnector<IIncomingPhysicalMessageContext, IAuditContext> forkConnector, OutgoingMessage message, string auditAddress, IIncomingPhysicalMessageContext sourceContext)
         {
+            Guard.AgainstNull(nameof(sourceContext), sourceContext);
```

Comment:
I didn't add guards for `message` and `auditAddress` here because `AuditContext` does that, with exactly the same parameter names.